### ISSUE TITLE
(2Beta3) Script Bug

### ISSUE THREAD

**Issue by ELV13Stm** [chrismaltby/gb-studio#579]
**Describe the bug**
Scripts in an Event Group sometimes doesn't do anything when copied and pasted in another Scene

**To Reproduce**
1. Create a BG Scene
2. Make a long script inside an Event Group in BG's On Init tab 
3. Copy and paste the script on another new BG's On Init tab
4. Play Test Game exit BG1 enter BG 2
5. Test to see if script should do anything on both scenes.

**Expected behavior**
sometimes the script won't do anything when entered a new BG scene, vice versa.

`**Screenshots**
![Classicvania Project 30-Sep-20 7_13_44 PM](https://user-images.githubusercontent.com/71689133/94678624-5f3d0180-0351-11eb-8ecf-4306ccd510ca.png)
The long script such as a player controller (move, idle attack, jump, crouch, special ability, pause menu) is copied and pasted onto every scene, now sometimes while play testing the script won't work like the attack or walking script

**Platform (please complete the following information):**
 - OS: Windows 10
 - App version GB Studio 2 beta 3 Dev Build

**Comment by RichardULZ** [comment]
Odd that you are using a timerscript to re set every attached script every 16 frames.

When you say sometimes the script won't do anything switching scenes, are you talking about the scene init script? Or just the buttons being attached? 
Do you have something visible at the end of init to confirm if init has run or failed regardless of attached buttons?

I have a suspicion holding or pressing a button while it is attached, and causing a scene switch, might somehow interupt the next init?

That or more likely it is a timerscript counter error where any actor with update script, and no wait, could cause timer to never count down / fire timerscript. #555

**Comment by ELV13Stm** [comment]
The script has sprite player change if you press left or right to a walking player sprite and change it idle when not pressed, same goes for attacking . Attached a weapon: item/projectile on A while changing sprite to an attack sprite sheet. It's copied and pasted ton every on init scene. The cue is that the attack thing sometimes won't work  whenever I switch scenes when its script is copied and pasted on that scene.

**Comment by RichardULZ** [comment]
But the attack/all of this is only ever assigned by a timerscript set on init? 
Very likely a problem with the timer script not counting/firing, rather than init failing.

**Comment by ELV13Stm** [comment]
Yes its all in a timer script, what do you think is a better script alt for it then?
